RULES/cc/toolchain.go

```diff
@@ -87,95 +97,52 @@ func (gcc GccToolchain) Freestanding() bool {
-}
-
-// StaticLibrary generates the command to build a static library.
-func (gcc GccToolchain) StaticLibrary(out core.Path, objs []core.Path) string {
-	// ar updates an existing archive. This can cause faulty builds in the case
-	// where a symbol is defined in one file, that file is removed, and the
-	// symbol is subsequently defined in a new file. That's because the old object file
-	// can persist in the archive. See https://github.com/daedaleanai/dbt/issues/91
-	// There is no option to ar to always force creation of a new archive; the "c"
-	// modifier simply suppresses a warning if the archive doesn't already
-	// exist. So instead we delete the target (out) if it already exists.
-	return fmt.Sprintf(
-		"rm %q 2>/dev/null ; %q rcs %q %s",
-		out,
-		gcc.Ar,
-		out,
-		joinQuoted(objs))
-}
-
-// SharedLibrary generates the command to build a shared library.
-func (gcc GccToolchain) SharedLibrary(out core.Path, objs []core.Path) string {
-	return fmt.Sprintf(
-		"%q -pipe -shared -o %q %s",
-		gcc.Cxx,
-		out,
-		joinQuoted(objs))
-}
-
-// Binary generates the command to build an executable.
-func (gcc GccToolchain) Binary(out core.Path, objs []core.Path, alwaysLinkLibs []core.Path, libs []core.Path, flags []string, script core.Path) string {
-	flags = append(gcc.LinkerFlags, flags...)
-	if script != nil {
-		flags = append(flags, "-T", fmt.Sprintf("%q", script))
-	} else if gcc.LinkerScript != nil {
-		flags = append(flags, "-T", fmt.Sprintf("%q", gcc.LinkerScript))
-	}
+func (gcc GccToolchain) Link() string {
+	return fmt.Sprintf("%q", gcc.Cc)
+}
 
-	return fmt.Sprintf(
-		"%q -pipe -o %q %s -Wl,-whole-archive %s -Wl,-no-whole-archive %s %s",
-		gcc.Cxx,
-		out,
-		joinQuoted(objs),
-		joinQuoted(alwaysLinkLibs),
-		joinQuoted(libs),
-		strings.Join(flags, " "))
+func (gcc GccToolchain) Assembler() string {
+	return fmt.Sprintf("%q", gcc.As)
 }
 
-// BlobObject creates an object file from any binary blob of data
-func (gcc GccToolchain) BlobObject(out core.OutPath, src core.Path) string {
-	return fmt.Sprintf(
-		"%q -r -b binary -o %q %q",
-		gcc.Ld,
-		out,
-		src)
+func (gcc GccToolchain) Archiver() string {
+	return fmt.Sprintf("%q", gcc.Ar)
 }
 
-// RawBinary strips ELF metadata to create a raw binary image
-func (gcc GccToolchain) RawBinary(out core.OutPath, elfSrc core.Path) string {
-	return fmt.Sprintf(
-		"%q -O binary %q %q",
-		gcc.Objcopy,
-		elfSrc,
-		out)
+func (gcc GccToolchain) ObjcopyCommand() string {
+	return fmt.Sprintf("%q", gcc.Objcopy)
+}
+
+func (gcc GccToolchain) CFlags() []string {
+	return gcc.CCompilerFlags
+}
+
+func (gcc GccToolchain) CxxFlags() []string {
+	return gcc.CxxCompilerFlags
+}
+
+func (gcc GccToolchain) AsFlags() []string {
+	return gcc.AsCompilerFlags
+}
+
+func (gcc GccToolchain) LdFlags() []string {
+	return gcc.LinkerFlags
+}
+
+func (gcc GccToolchain) NewWithStdLib(includes []core.Path, deps []Dep, linkerScript core.Path, toolchainName string) GccToolchain {
+	gcc.Includes = includes
```

Comment:
These are not used anywhere.

Reply:
Maybe that is fine (in that we don't actually need to have includes with a `Toolchain`). But in that case we should get rid of the member altogether.